src/layout/dom.ts

```diff
@@ -41,15 +59,20 @@ export function add(node: Node, data: NodeInfo, source: Source): void {
     let nextId = getNextId(node);
     let masked = true;
     let parent = null;
+    let region = regionMap.has(node) ? regionMap.get(node) : null;
 
     if (parentId >= 0 && values[parentId]) {
         parent = values[parentId];
         parent.children.push(id);
+        region = region === null ? parent.region : region;
         masked = parent.metadata.masked;
     }
 
     if (data.attributes && Constant.MASK_ATTRIBUTE in data.attributes) { masked = true; }
     if (data.attributes && Constant.UNMASK_ATTRIBUTE in data.attributes) { masked = false; }
+    if (data.attributes && Constant.CLARITY_ID_ATTRIBUTE in data.attributes) {
+        regionMap.set(node, data.attributes[Constant.CLARITY_ID_ATTRIBUTE]);
```

Comment:
>CLARITY_ID_ATTRIBUTE [](start = 53, length = 20)

What is this and it's not thee "id" attribute, right?

Reply:
It is data-clarity tag. I will change it to data-clarity-region to be consistent with rest of the code. I will make the change.

Reply:
Named it data-clarity-region and CLARITY_REGION_ATTRIBUTE